### ISSUE TITLE
Can I give my own names to people?

### ISSUE THREAD

**Issue by NickBarnes** [keybase/client#14406]
I want the people I know to appear using their names, not whatever random KeyBase username they've chosen or been assigned. This doesn't seem possible in the current UI (2.8.0-20181023171250+c7287208b0 on macOS, 2.7.4-20181009145447+8dd7ab9792 on iOS).

**Comment by DonaldKBrown** [comment]
@NickBarnes This seems possible in the most recent client (on Linux, at least). I tested by picking a random person in the "Consider following..." reel and searching for their name. It actually turned up two possible results, one with a much different username but very similar real name:
![Searched for Mat Schaffer](https://donaldkbrown.keybase.pub/Screenshots/names.png)

**Comment by mossy-nw** [comment]
I think what @NickBarnes is requesting is the capability for a person's client to assign local nicknames -- synchronized across all devices -- to their individual contacts (long possible with many still-popular old-school XMPP chat clients, etc.).

Such local nicknames would be very useful for people with numerous contacts who use pseudonymous usernames, especially important given the (global) immutability of usernames.

Since this would only happen (and also only happen optionally) on an individual's client(s), such a feature should not interfere with the security property of globally-unique usernames?

**Comment by NickBarnes** [comment]
Yes, exactly. I might know only one Fred Smith, so I'd like to refer to him as Fred Smith but he has the username @spondulix3745 or some such nonsense (either forced because of global uniqueness or as a personal preference). I want to attach the real name "Fred Smith" to that username. Thereafter, I want him to appear as such in the user interface: the fact his account has some other name is irrelevant to me. In the ideal world (as provided by many existing chat clients such as WhatsApp) I could refer to him as @ Fred Smith in actual messages, the user interface will automagically translate that into his account username on the way out, so every user in the chat will see that reference as their own local nickname for his account, if they have one, or as spondulix3745 if they do not.
